src/librustc_resolve/lib.rs

```diff
@@ -3672,27 +3703,40 @@ impl<'a> Resolver<'a> {
                         Some(ModuleOrUniformRoot::Module(module)) => module.def(),
                         _ => None,
                     };
-                    let msg = if module_def == self.graph_root.def() {
+                    let (label, suggestion) = if module_def == self.graph_root.def() {
                         let is_mod = |def| match def { Def::Mod(..) => true, _ => false };
                         let mut candidates =
                             self.lookup_import_candidates(ident, TypeNS, is_mod);
                         candidates.sort_by_cached_key(|c| {
                             (c.path.segments.len(), c.path.to_string())
                         });
                         if let Some(candidate) = candidates.get(0) {
-                            format!("did you mean `{}`?", candidate.path)
+                            (
+                                String::from("unresolved import"),
+                                Some((
+                                    ident.span,
+                                    String::from("a similar path resolves"),
```

Comment:
```suggestion
                                    String::from("a similar path exists"),
```

Word choice: I think "exists" feels less stilted than "resolves" here.